pkgs/applications/networking/browsers/firefox/common.nix

```
@@ -140,148 +166,150 @@ buildStdenv.mkDerivation ({
 
   inherit src unpackPhase meta;
 
+  # add profiling and a second configure phase if we build with pgo
+  phases = lib.optionals pgoSupport [
+    "unpackPhase"
+    "patchPhase"
+    "configurePhase"
+    "buildPhase"
+    "profilingPhase"
+    "configurePhase"
+    "buildPhase"
+    "fixupPhase"
+    "installPhase"
+    "installCheckPhase"
+  ];
```

It's generally not a good idea to directly set `phases`. This may cause issue when the default phases change.

```suggestion
  # Add another configure-build-profiling run before the main configure phase.
  preConfigurePhases = lib.optionals pgoSupport [
    "configurePhase"
    "buildPhase"
    "profilingPhase"
  ];
```

---

Interesting, I didn't know.